### ISSUE TITLE
Simplifications with lerp are missed

### ISSUE THREAD

**Issue by dsharletg** [halide/Halide#3857]
I have some cases where code uses a manually written lerp (that is not as good as Halide's lerp because it does not use the nice math tricks to avoid widening arithmetic) because it wants to specialize the code that lead to special cases of interpolation factors. The simplifier simplifies these when the lerp is written using Halide expressions, but the lerp intrinsic does not get as many simplifications.

I think maybe lerp should be lowered to expressions earlier?

**Comment by zvookin** [comment]
Didn't we add the lerp intrinsic to be able to codegen it more easily and
directly for GPU? If so, then we probably need to make the lowering happen
at different places for different targets. That seems like a reasonable
thing to do to me.

-Z-

**Comment by shoaibkamil** [comment]
lerp is lowered during codegen; it may be fruitful to add simplification rules specific to lerp. (cc @abadams)

**Comment by abadams** [comment]
I agree with Zalman - lowering lerp for cpu targets could easily be done before the last simplification.

Not that I'm opposed to lerp-specific rules. I guess they may benefit gpu targets. Is lerp just for opengl at this point?

**Comment by shoaibkamil** [comment]
It's just for OpenGL, I believe, but it's on my to-do list to add for Metal.
